src/Extension/CoreExtension.php

```diff
@@ -1644,7 +1644,7 @@ public static function getAttribute(Environment $env, Source $source, $object, $
 
         // array
         if (Template::METHOD_CALL !== $type) {
-            $arrayItem = \is_bool($item) || \is_float($item) ? (int) $item : $item = (string) $item;
+            $arrayItem = \is_bool($item) || \is_float($item) ? (int) $item : $item;
```

Comment:
@nicolas-grekas I'm not sure if there was a good reason for the string casting. So maybe it should be done somewhere else? Or is it ok to just remove the casting?